modules/rostests/apitests/shell32/SHShouldShowWizards.cpp

```diff
@@ -0,0 +1,128 @@
+
+class CDummyClass
+    : public IServiceProvider
+    , public IShellBrowser
+{
+public:
+    CDummyClass() { }
+
+    IUnknown *get_unknown()
+    {
+        return static_cast<IServiceProvider *>(this);
+    }
+
+    // *** IUnknown methods ***
+    STDMETHODIMP QueryInterface(REFIID riid, VOID **ppvObj) override
+    {
+        if (riid == IID_IUnknown || riid == IID_IServiceProvider)
+        {
+            AddRef();
+            *ppvObj = static_cast<IServiceProvider *>(this);
+            return S_OK;
+        }
+        return E_NOINTERFACE;
+    }
+    STDMETHODIMP_(ULONG) AddRef() override
+    {
+        return 1;
+    }
+    STDMETHODIMP_(ULONG) Release() override
+    {
+        return 1;
+    }
+
+    // *** IOleWindow methods ***
+    STDMETHODIMP GetWindow(HWND *phwnd) override { return E_NOTIMPL; }
+    STDMETHODIMP ContextSensitiveHelp(BOOL fEnterMode) override { return E_NOTIMPL; }
+
+    // *** IShellBrowser methods ***
+    STDMETHODIMP InsertMenusSB(HMENU hmenuShared, LPOLEMENUGROUPWIDTHS lpMenuWidths) override { return E_NOTIMPL; }
+    STDMETHODIMP SetMenuSB(HMENU hmenuShared, HOLEMENU holemenuRes, HWND hwndActiveObject) override { return E_NOTIMPL; }
+    STDMETHODIMP RemoveMenusSB(HMENU hmenuShared) override { return E_NOTIMPL; }
+    STDMETHODIMP SetStatusTextSB(LPCOLESTR pszStatusText) override { return E_NOTIMPL; }
+    STDMETHODIMP EnableModelessSB(BOOL fEnable) override { return E_NOTIMPL; }
+    STDMETHODIMP TranslateAcceleratorSB(MSG *pmsg, WORD wID) override { return E_NOTIMPL; }
+    STDMETHODIMP BrowseObject(LPCITEMIDLIST pidl, UINT wFlags) override { return E_NOTIMPL; }
+    STDMETHODIMP GetViewStateStream(DWORD grfMode, IStream **ppStrm) override { return E_NOTIMPL; }
+    STDMETHODIMP GetControlWindow(UINT id, HWND *lphwnd) override { return E_NOTIMPL; }
+    STDMETHODIMP SendControlMsg(UINT id, UINT uMsg, WPARAM wParam, LPARAM lParam, LRESULT *pret) override { return E_NOTIMPL; }
+    STDMETHODIMP QueryActiveShellView(struct IShellView **ppshv) override { return E_NOTIMPL; }
+    STDMETHODIMP OnViewWindowActive(struct IShellView *ppshv) override { return E_NOTIMPL; }
+    STDMETHODIMP SetToolbarItems(LPTBBUTTON lpButtons, UINT nButtons, UINT uFlags) override { return E_NOTIMPL; }
+
+    // *** IServiceProvider methods ***
+    STDMETHODIMP QueryService(REFGUID guidService, REFIID riid, void **ppvObject) override
+    {
+        if (riid == IID_IShellBrowser)
+        {
+            AddRef();
+            *ppvObject = static_cast<IShellBrowser *>(this);
+            return S_OK;
+        }
+        return E_FAIL;
+    }
+};
+
+static VOID SetShowWizardsTEST(BOOL bValue)
+{
+    DWORD dwValue = bValue;
+    SHRegSetUSValueW(L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Advanced",
+                     L"ShowWizardsTEST", REG_DWORD, &dwValue, sizeof(dwValue), SHREGSET_FORCE_HKCU);
+}
+
+START_TEST(SHShouldShowWizards)
+{
+    if (IsWindowsVistaOrGreater())
+    {
+        skip("Vista+\n");
```

Comment:
Why? If I understand correctly https://www.geoffchappell.com/studies/windows/shell/shell32/api/index.htm?tx=68 the SHShouldShowWizards function is really exported in Vista+.
(I've checked on Win7 x64 and it's exported, but **NOT** in named form: it's exported via ordinal 237. And this is the same on Windows 2003 x86 where it's also at the same ordinal.)

Reply:
It didn't work as I expected.
Under investigation.

Reply:
Afaics it also has one single parameter.

Reply:
Fixed for Vista+.